dev/io.openliberty.microprofile.metrics.internal.3.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/MetricsJaxRsEMCallbackImpl.java

```diff
@@ -57,14 +61,39 @@ public synchronized static Counter registerOrRetrieveRESTUnmappedExceptionMetric
 		
 		return counter;
 	}
-	
+
 	@Override
-	public Map<String, Object> onDefaultMappedException(Throwable t, int statusCode, ResourceInfo resourceInfo) {
-		Map.Entry<String, String> classXmethod = resolveSimpleTimerClassMethodTags(resourceInfo);
+	public Map<String, Object> onDefaultMappedException(Throwable throwable, int statusCode, ResourceInfo resourceInfo) {
+
+		StackTraceElement[] ste = throwable.getStackTrace();
+
+		/*
+		 * If the Exception originates from from the Metrics CDI Bundle we do not want to count
+		 * the exception.
+		 * 
+		 * Validate by checking the first element on the stack to see if it came from the two packages
+		 * in the Metrics CDI bundle that throws Exceptions.
+		 */
+		if (!(ste[0].getClassName().startsWith(metriCDIBundles[0]) || ste[0].getClassName().startsWith(metriCDIBundles[1]))) {
+			Map.Entry<String, String> classXmethod = resolveSimpleTimerClassMethodTags(resourceInfo);
+			registerOrRetrieveRESTUnmappedExceptionMetric(classXmethod.getKey() ,classXmethod.getValue()).inc();
+		}
 
-		registerOrRetrieveRESTUnmappedExceptionMetric(classXmethod.getKey() ,classXmethod.getValue()).inc();
+		/*
+		 * Will print out the stack trace as one string.
+		 * This is chosen over the alternative which is
+		 * throwable.printStackTrace() which will stream
+		 * every stack element/line as a logging event making
+		 * messages.log cluttered. The trade-off being that
+		 * internal classes are not hidden in the console
+		 * output.
+		 */
+		StringWriter sw = new StringWriter();
+		PrintWriter pw = new PrintWriter(sw);
+		throwable.printStackTrace(pw);
+		System.err.print(sw.toString());
```

Comment:
suggest you just use throwable.printStackTrace -- we have another issue coming soon which will handle merging stack traces into a single event